BrainPortal/app/controllers/neurohub_application_controller.rb

```diff
@@ -83,6 +84,35 @@ def check_if_locked
 
   end
 
+  ########################################################################
+  # Messaging System Filters (presently only invite acceptance)
+  ########################################################################
+
+  # Find new messages to be displayed at the top of the page.
+  def prepare_messages #:nodoc:
```

Comment:
This is a complete duplication of the method of the same name in Application controller, except for a single line. That is not DRY at all.

Please:

1. remove this method here since it's already defined in ApplicationController
2. change the method in ApplicationController such that the one line that differs invokes a method that can be overrided, which you will add to both files:
```ruby
   unread_messages = unread_messages_to_display
```

so that  in ApplicationController:

```ruby
# put this method just after prepare_messages
def unread_messages_to_display 
   current_user.messages.where( :read => false ).order( "last_sent DESC" )
end
```

and in NeurohubApplicationController we ONLY have this now:

```ruby
def unread_messages_to_display
  current_user.messages.where( :read => false, :header => 'Invitation Accepted' ).order( "last_sent DESC" )
end
```

Reply:
added